node/src/reactor/validator.rs

```diff
@@ -450,9 +450,9 @@ impl reactor::Reactor for Reactor {
             chainspec_loader.chainspec().as_ref(),
         )?;
         let mut effects = reactor::wrap_effects(Event::BlockProposer, block_proposer_effects);
-        let genesis_state_root_hash = chainspec_loader.genesis_state_root_hash();
         let block_executor = BlockExecutor::new(
-            genesis_state_root_hash,
+            chainspec_loader.starting_state_root_hash(),
+            chainspec_loader.highest_block_header(),
```

Comment:
Is it correct to initialize `BlockExecutor` with the same values for both joining and validator reactors? After joining (in the validator reactor) both `initial_state_root_hash` and `highest_block_header` might change.

Reply:
This is the best we have. But note that below we call `with_parent_map` on it - which should be bringing it up to date with whatever we downloaded during syncing.